src/AnyAddress.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "AnyAddress.h"
+
+#include <utility>
+#include "Coin.h"
+
+namespace TW {
+
+Data AnyAddress::getData() const {
+    Data data;
+    try {
+        data = TW::addressToData(coin, address);
+    } catch (...) {}
```

Comment:
Let's catch the exception directly

Reply:
Actually, I didn't find any exceptions thrown, so I removed it completely

Reply:
I think that's dangerous and not correct.  addressToData() calls into each chains's Entry method, so there are many to check. There usually they create an Address instance from the string, and Address constructors often throw on invalid data.
Please leave the catch, it's safer.
I think you should be able to find a test case with a coin and invalid address string that throws here.